src/api/app/controllers/webui/users/notifications_controller.rb

```diff
@@ -62,4 +63,9 @@ def projects_for_filter
            .order('name desc').group(:name).count # this query returns a sorted-by-name hash like { "home:b" => 1, "home:a" => 3  }
            .sort_by(&:last).reverse.to_h # this sorts the hash by amount: { "home:a" => 3, "home:b" => 1 }
   end
+
+  def notifications_count
+    counted_notifications = NotificationsFinder.new(User.session.notifications.for_web).unread.group(:notifiable_type).count
+    counted_notifications.merge!('unread' => User.session.unread_notifications)
```

Comment:
Since `'unread'` is always present, this has the side-effect of displaying a badge even when there are no unread notifications.

![unnecessary-badge](https://user-images.githubusercontent.com/1102934/85288201-a9cafa00-b495-11ea-83a4-84119ffa4c2e.png)

Reply:
Ah right, going to address this

Reply:
I'm checking now for `zero?` in the helper @dmarcoux